src/managed/dir.proj

```diff
@@ -3,11 +3,15 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
 
-  <Target Name="Build">
+  <Target Name="Build"
+          DependsOnTargets="GetLatestCommitHash">
     <MakeDir  Condition="!Exists('$(pOutDir)')" Directories="$(pOutDir)" />
 
     <PropertyGroup>
       <Args>/p:Configuration=$(ConfigurationGroup)</Args>
+      <_EnvironmentVariables>NUGET_PACKAGES=$(PackagesDir)</_EnvironmentVariables>
```

Comment:
Is there any reason we cannot pass these parameters as args?

As for the PackagesDir I think we should be switching to using a nuget.config file to define the package folder instead of relying on the NUGET_PACKAGES env variable.